### PR TITLE
misc: Spring dependencies update

### PR DESCRIPTION
**Pull request by kamilogorek**
Encoding update is based on https://docs.rs/crate/percent-encoding/1.0.1/source/lib.rs
`uuid` cannot be updated due to `proguard` `symbolic-common` and `debugid` depending on version `0.7`.

### CODE REVIEW

File: src/api.rs [hunk 1]
```diff
@@ -43,6 +43,18 @@ use crate::utils::sourcemaps::get_sourcemap_reference_from_headers;
 use crate::utils::ui::{capitalize_string, make_byte_progress_bar};
 use crate::utils::xcode::InfoPlist;
 
+const QUERY_ENCODE_SET: &AsciiSet = &CONTROLS.add(b' ').add(b'"').add(b'#').add(b'<').add(b'>');
```
Comment (jan-auer) [inline src/api.rs]: Isn't this missing the condition from `SIMPLE_ENCODE_SET`?

```
byte < 0x20 || byte > 0x7E
```

Comment (jan-auer) [reply src/api.rs]: Nvm, this is in `CONTROLS`.

File: src/api.rs [hunk 2]
```diff
@@ -43,6 +43,18 @@ use crate::utils::sourcemaps::get_sourcemap_reference_from_headers;
...
+const DEFAULT_ENCODE_SET: &AsciiSet = &CONTROLS
```
Comment (jan-auer) [inline src/api.rs]: Could you add to `QUERY_ENCODE_SET` instead of repeating?

Comment (kamilogorek) [reply src/api.rs]: Wasn't sure if it's the same reference or does builder always return new instance.

**Pull request merged**